subsys/net/lib/dns/mdns_responder.c

```diff
@@ -72,6 +75,33 @@ static void create_ipv4_addr(struct sockaddr_in *addr)
 	/* Well known IPv4 224.0.0.251 address */
 	addr->sin_addr.s_addr = htonl(0xE00000FB);
 }
+#endif /* CONFIG_NET_IPV4 */
+
+static int setup_dst_addr(struct net_context *ctx,
+			  struct net_pkt *pkt,
+			  struct sockaddr *dst,
+			  socklen_t *dst_len)
+{
+	if (net_pkt_family(pkt) == AF_INET) {
```

Comment:
The changes look good but it would be better to use `IS_ENABLED()` macro here so that we get all the code paths compile tested.
So like this:
```
if (IS_ENABLED(CONFIG_NET_IPV4) && net_pkt_family(pkt) == AF_INET) {
...
} else if (IS_ENABLED(CONFIG_NET_IPV6) && net_pkt_family(pkt) == AF_INET6) {
...
} else {
	return -EPFNOSUPPORT;
}
```
Then you can also remove the ifdefs around `create_ipv4_addr` and `create_ipv6_addr`

Reply:
Hey good point, thanks, done.